pkg/auth/authproviders/registry_httphandler.go

```diff
@@ -279,9 +279,10 @@ func (r *registryImpl) providersHTTPHandler(w http.ResponseWriter, req *http.Req
 		r.error(w, err, typ, clientState, testMode)
 		return
 	}
+
 	userRoles := userInfo.GetRoles()
 	if len(userRoles) == 0 {
-		err := errox.GenericNoValidRole()
+		err := errox.NotAuthorized.CausedBy("no valid role for the user")
```

Comment:
How about adding explicit information about missing roles. `no valid role` sounds like there are some roles but not valid for this situation
```suggestion
		err := errox.NotAuthorized.CausedBy(fmt.Sprintf("%s has no roles", userInfo.GetFriendlyName()))
```